### PR TITLE
Encrypt credentials

### PR DESCRIPTION
**Pull request by mushtaqak**
Encrypt transcript organization credentials using fernet keys.

[EDUCATOR-1392](https://openedx.atlassian.net/browse/EDUCATOR-1392)

[Related platform PR](https://github.com/edx/edx-platform/pull/16201)

### CODE REVIEW

File: VEDA/django_secrets.py [hunk 1]
```diff
@@ -19,6 +19,8 @@
         return_dict = None
 
 
+# Don't share with anybody.
+FERNET_KEYS = return_dict['fernet_keys'] or ['test-ferent-key']
```
Comment (muzaffaryousaf) [inline VEDA/django_secrets.py]: @mushtaqak can you confirm with @muhammad-ammar . I think we decided to remove django_secrets.py

File: instance_config.yaml [hunk 2]
```diff
@@ -21,6 +21,9 @@ DATABASES:
 
 django_secret_key: ""
 
+# Fernet keys
+fernet_keys: []
```
Comment (muzaffaryousaf) [inline instance_config.yaml]: do add in test_config as well.

File: VEDA_OS01/models.py [hunk 3]
```diff
@@ -664,9 +665,9 @@ class TranscriptCredentials(TimeStampedModel):
         help_text='This value must match the value of organization in studio/edx-platform.'
     )
     provider = models.CharField('Transcript provider', max_length=50, choices=TranscriptProvider.CHOICES)
-    api_key = models.CharField('API key', max_length=255)
-    api_secret = models.CharField('API secret', max_length=255, null=True, blank=True)
-
+    api_key = EncryptedTextField('API key', max_length=255)
```
Comment (muzaffaryousaf) [inline VEDA_OS01/models.py]: Do we need to make it null as well ?

Comment (mushtaqak) [reply VEDA_OS01/models.py]: No, API Key is must.

File: test_config.yaml [hunk 4]
```diff
@@ -36,6 +36,9 @@ val_video_transcript_status_url: http://val.edx.org/video/status
 celery_worker_queue: encode_worker
 celery_deliver_queue: deliver_worker
 
+# Fernet keys
+FERNET_KEYS: ['test-current-key', 'old-key-1', 'old-key-2']
```
Comment (muhammad-ammar) [inline test_config.yaml]: Will this list keeps populated forever?

File: VEDA_OS01/tests/test_models.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class TranscriptCredentialsModelTest(TransactionTestCase):
+    """
+    Transcript credentials model tests
+    """
+        
+    def test_credential_keys(self):
+        """
+        Tests transcrip credential keys are stored correctly.
```
Comment (muhammad-ammar) [inline VEDA_OS01/tests/test_models.py]: typo: `transcrip` -> `transcript`

File: test_config.yaml (hunk 4 above)
Comment (muhammad-ammar) [reply test_config.yaml]: Also how the rotation of key will work? What is the best practice or pattern?

File: VEDA_OS01/tests/test_models.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
+""" Model tests """
+from django.test.testcases import TransactionTestCase
+from VEDA_OS01.models import TranscriptCredentials, TranscriptProvider
+
+
+class TranscriptCredentialsModelTest(TransactionTestCase):
```
Comment (muhammad-ammar) [inline VEDA_OS01/tests/test_models.py]: I think we also need tests for key rotation.

File: test_config.yaml (hunk 4 above)
Comment (muhammad-ammar) [inline test_config.yaml]: Why changes in `test_config.yaml`? We are not using this test config for `test_models.py`?

**Comment by mushtaqak** [comment]
@muhammad-ammar @Qubad786 feedback addressed

**Pull request merged**